editor/import/resource_importer_wav.cpp

```diff
@@ -174,7 +174,7 @@ Error ResourceImporterWAV::import(const String &p_source_file, const String &p_s
 			file->get_16(); // block align (unused)
 			format_bits = file->get_16(); // bits per sample
 
-			if (format_bits % 8) {
+			if (format_bits % 8 || format_bits == 0) {
 
 				ERR_PRINT("Strange number of bits in sample (not 8,16,24,32)");
```

Comment:
Might be worth improving the English in the error while at it. "Invalid amount of bits in the sample (should be one of 8, 16, 24 or 32).".

I also wonder if it should not return instead of breaking.

Reply:
Let me know if I should change it to return instead. The imported wav is almost identical to `AudioStreamSample.new()`.

Reply:
IMO it should return with `ERR_FAIL_COND_V(ERR_INVALID_DATA)` (and thus you can use `ERR_EXPLAIN` instead of `ERR_PRINT`).

Same here: https://github.com/godotengine/godot/blob/b884559393b60a8003df5c468ed0deef1e27eb75/editor/import/resource_importer_wav.cpp#L192

(And the comment should be fixed, it's "DATA CHUNK".)

Reply:
IMO it should return on all the `break`s in the format chunk, as they all imply invalid format. And then return on the first `break` in the data chunk, which also implies invalid data (currently it will process data using `format_bits == 0` which is the init value).